tests/test_check_parallel.py

```diff
@@ -0,0 +1,141 @@
+import os
+import shutil
+import unittest
+from base import *
+
+
+class TestCheckParallel(TestZstash):
+    """
+    Test `zstash check` in parallel.
+    """
+    # `zstash check` is tested in TestCheck and TestCheckParallel.
+    # x = on, no mark = off, b = both on and off tested
+    # option | Check | CheckMismatch | KeepTars | CheckParallel | CheckParallelVerboseMismatch | KeepTarsWithPreviouslySetHPSS |
+    # --hpss    |x|x|x|x|x| |
+    # --workers | | | |x|x|x|
+    # --keep    | | | | | | |
+    # -v        |b|x| |b|x| |
+
+    def helperCheckParallel(self, test_name, hpss_path, zstash_dir=ZSTASH_DIR, verbose=False):
+        """
+        Test `zstash check` in parallel.
+        """
+        self.hpss_path = hpss_path
+        use_hpss = self.setupDirs(test_name)
+        self.create(use_hpss, zstash_dir)
+        self.add_files(use_hpss, zstash_dir)
+        self.extract(use_hpss, zstash_dir)
+        print_starred('Checking the files in parallel.')
+        self.assertWorkspace()
+        os.chdir(self.test_dir)
+        v_option = ' -v' if verbose else ''
+        cmd = '{}zstash check{} --hpss={} --workers=3'.format(zstash_dir, v_option, self.hpss_path)
+        output, err = run_cmd(cmd)
+        os.chdir(TOP_LEVEL)
+        expected_present = ['Checking file0.txt', 'Checking file0_hard.txt', 'Checking file0_soft.txt',
+                            'Checking file_empty.txt', 'Checking dir/file1.txt', 'Checking empty_dir',
+                            'Checking dir2/file2.txt', 'Checking file3.txt', 'Checking file4.txt', 'Checking file5.txt']
+        expected_absent = ['ERROR']
+        self.check_strings(cmd, output + err, expected_present, expected_absent)
+
+    def helperCheckParallelVerboseMismatch(self, test_name, hpss_path, zstash_dir=ZSTASH_DIR):
+        """
+        Test `zstash check -v` in parallel with MD5 mismatch.
+        """
+        self.hpss_path = hpss_path
+        use_hpss = self.setupDirs(test_name)
+        self.create(use_hpss, zstash_dir)
+        self.add_files(use_hpss, zstash_dir)
+        self.extract(use_hpss, zstash_dir)
+        print_starred('Causing MD5 mismatch errors and checking the files in parallel.')
+        self.assertWorkspace()
+        os.chdir(self.test_dir)
+        shutil.copy('zstash/index.db', 'zstash/index_old.db')
+        print('Messing up the MD5 of all of the files with an even id.')
+        cmd = ['sqlite3', 'zstash/index.db', 'UPDATE files SET md5 = 0 WHERE id % 2 = 0;']
```

Comment:
@golaz What's the `id`? Is it randomly generated or does it depend on archiving order? Looks like odd and even file names are mixed in `expected_present` and `expected_absent`.